jobs/dataproc_scripts/mt_to_es.py

```diff
@@ -0,0 +1,695 @@
+                    'index.mapping.total_fields.limit': 10000,
+                    'index.refresh_interval': -1,
+                    'index.codec': 'best_compression',  # halves disk usage, no difference in query times
+                },
+            }
+
+            logging.info(
+                'create_mapping - elasticsearch schema: \n'
+                + pformat(elasticsearch_schema)
+            )
+            logging.info('==> creating elasticsearch index {}'.format(index_name))
+
+            self.es.indices.create(index=index_name, body=body)
+        else:
+            if create_only:
+                raise ValueError('Index {} already exists'.format(index_name))
+
+            logging.info(
+                '==> updating elasticsearch index {}. New schema:\n{}'.format(
+                    index_name, pformat(elasticsearch_schema)
+                )
+            )
+
+            self.es.indices.put_mapping(index=index_name, body=index_mapping)
+
+    def route_index_to_temp_es_cluster(self, index_name):
+        """
+        Apply shard allocation filtering rules to route the given index to elasticsearch loading nodes
+        """
+        self._update_settings(
+            index_name,
+            {
+                'index.routing.allocation.require._name': LOADING_NODES_NAME,
+                'index.routing.allocation.exclude._name': '',
+            },
+        )
+
+    def route_index_off_temp_es_cluster(self, index_name):
+        """
+        Move any shards in the given index off of loading nodes
+        """
+        self._update_settings(
+            index_name,
+            {
+                'index.routing.allocation.require._name': '',
+                'index.routing.allocation.exclude._name': LOADING_NODES_NAME,
+            },
+        )
+
+    def _update_settings(self, index_name, body):
+        logging.info('==> Setting {} settings = {}'.format(index_name, body))
+
+        self.es.indices.put_settings(index=index_name, body=body)
+
+    def _get_index_meta(self, index_name):
+        mappings = self.es.indices.get_mapping(index=index_name)
+        return mappings.get(index_name, {}).get('mappings', {}).get('_meta', {})
+
+    def wait_for_shard_transfer(self, index_name, num_attempts=1000):
+        """
+        Wait for shards to move off of the loading nodes before connecting to seqr
+        """
+        for _ in range(num_attempts):
+            shards = self.es.cat.shards(index=index_name)
+            if LOADING_NODES_NAME not in shards:
+                logging.warning('Shards are on {}'.format(shards))
+                return
+            logging.warning(
+                'Waiting for {} shards to transfer off the es-data-loading nodes: \n{}'.format(
+                    len(shards.strip().split('\n')), shards
+                )
+            )
+            time.sleep(5)
+
+        raise Exception('Shards did not transfer off loading nodes')
+
+    def export_table_to_elasticsearch(
```

Comment:
Is this mostly a copy from the `hail-elasticsearch-pipelines` code? Would it be too painful to import that instead of duplicating it here? Is there any risk this might diverge from upstream?

Reply:
Yeah, and I used to import it, but it felt that maintaining a fork of https://github.com/populationgenomics/hail-elasticsearch-pipelines with just added `setup.py` that builds a package with modules named `hail_scripts` and `lib`, and maintaining that package, is not ideal either. But agree that it would make it harder if we diverge, I'll think if I can maybe get away with a git submodule